daemon-scheduler/internal/features/steps/step_defns.go

```diff
@@ -52,6 +67,123 @@ func init() {
 		return
 	}
 
+	// TODO: Change these os.Exit calls to T.Errorf. Currently unable to do so because T is not initialized until the first test.
+	// (https://github.com/gucumber/gucumber/issues/28)
+	BeforeAll(func() {
+		clusterName := wrappers.GetClusterName()
+
+		_, err := ecsWrapper.CreateCluster(clusterName)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(errorCode)
+		}
+
+		err = terminateAllContainerInstances(ec2Wrapper, ecsWrapper, clusterName)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(errorCode)
+		}
+
+		taskDefinitionSleep300 := "esh_test_sleep300"
+		taskDefnARN, err = ecsWrapper.RegisterTaskDefinition(taskDefinitionSleep300)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(errorCode)
+		}
+
+		azs, err := ec2Wrapper.DescribeAvailabilityZones()
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(errorCode)
+		}
+
+		asg := wrappers.GetASGName()
+		if asg == defaultASGClusterName {
```

Comment:
If users wanna pass in their custom ASG, the test will pick up that ASG and not create a new ASG itself.